app/scripts/controllers/network/provider-api-tests/shared-tests.js

```diff
@@ -3204,56 +3308,464 @@ export function testsForRpcMethodSupportingBlockParam(
         });
       });
 
-      it('makes an additional request to the RPC endpoint if the given block number matches the latest block number', async () => {
-        await withMockedCommunications({ providerType }, async (comms) => {
-          const request = {
-            method,
-            params: buildMockParams({ blockParamIndex, blockParam: '0x100' }),
-          };
+      describe('less than the current block number', () => {
+        it('makes an additional request to the RPC endpoint', async () => {
+          await withMockedCommunications({ providerType }, async (comms) => {
+            const request = {
+              method,
+              params: buildMockParams({ blockParamIndex, blockParam }),
```

Comment:
Now I kind of wish I hadn't extracted this `blockParam`. It's a bit hard to see just by looking at this `describe` that it's _this_ value we care about being less than the current block number. Would it be clearer if we specified what `blockParam` was in the `less than the current block number`, `equal to the current block number`, and `greater than the current block number` tests? For instance:

``` suggestion
              params: buildMockParams({ blockParamIndex, blockParam: '0x100' }),
```

Reply:
Yeah I agree, this is hard to read. It feels wrong to ignore the loop variable as well though.

Maybe it will suffice for now to add a comment pointing out why this is relevant? And we can restructure this later to make more sense.

Reply:
Yeah, if it's a big deal to change this I'm fine with that solution.

Reply:
Well, not a big deal, but I just don't want the loop variable to become unused and potentially out-of-sync with what the test is doing. That would also be confusing. And a larger refactor would be beyond what I'd like to do in this PR.

That said, I did find a way to combine at least the first two of these blocks (the two longer ones), hopefully making this more readable as well. And I've added the comments. Done here: 38004d3655ea70bef8a52f3a0e4b3abfc369491a